trees/redblacktree/redblacktree.go

```diff
@@ -192,6 +192,11 @@ func (tree *Tree) Clear() {
 	tree.size = 0
 }
 
+// Return comparator of the tree
+func (tree *Tree) Comparator() utils.Comparator {
```

Comment:
Perhaps there is no need for a getter (conisdered evil nowadays). I've already exposed the _Root_, so simply expose _Compoarator and I can immediately merge the PR:
```go
type Tree struct {
	Root       *Node
	size       int
	Comparator utils.Comparator
}
```

Regarding the pointers, I'll go through them later and see if it could be optimized.